faiss/gpu/GpuIndexIVF.cu

```diff
@@ -426,18 +431,40 @@ void GpuIndexIVF::trainQuantizer_(idx_t n, const float* x) {
         return;
     }
 
-    if (this->verbose) {
-        printf("Training IVF quantizer on %ld vectors in %dD\n", n, d);
+    printf("Training IVF quantizer on %ld vectors in %dD\n", n, d);
+
+    if (config_.use_raft) {
+        printf("Using raft to train quantizer for %d vectors\n", n);
+        const raft::handle_t& raft_handle =
+                resources_->getRaftHandleCurrentDevice();
+
+        raft::neighbors::ivf_flat::index_params raft_idx_params;
+        raft_idx_params.n_lists = nlist;
+        raft_idx_params.metric = raft::distance::DistanceType::L2Expanded;
+        raft_idx_params.add_data_on_build = false;
+        raft_idx_params.kmeans_trainset_fraction = 1.0;
+        raft_idx_params.kmeans_n_iters = 100;
```

Comment:
You can use the same clustering parameters (`IndexIVF::cp`) for this, and also derive `kmeans_trainset_fraction` based on `cp.min_points_per_centroid`/`cp.max_points_per_centroid` and the number of vectors passed for training? Can also look to see if any of the other clustering parameters have analogues in RAFT.

https://github.com/facebookresearch/faiss/blob/main/faiss/Clustering.h#L22